.circleci/config.yml

```diff
@@ -41,10 +148,140 @@ jobs:
+          command: |
+            solc_version=$(jq --raw-output .version solc-js/package.json)
+
+            cd boilerplate/
+
+            sed -i 's|pragma solidity [^;]\+;|pragma solidity *;|g' contracts/Token.sol
+
+            {
+              echo "const {TASK_COMPILE_SOLIDITY_GET_SOLC_BUILD} = require('hardhat/builtin-tasks/task-names');"
+              echo "const assert = require('assert');"
+              echo
+              echo "subtask(TASK_COMPILE_SOLIDITY_GET_SOLC_BUILD, async (args, hre, runSuper) => {"
+              echo "    assert(args.solcVersion == '${solc_version}', 'Unexpected solc version: ' + args.solcVersion);"
+              echo "    return {"
+              echo "        compilerPath: '$(realpath "../solc-js/soljson.js")',"
+              echo "        isSolcJs: true,"
+              echo "        version: args.solcVersion,"
+              echo "        longVersion: args.solcVersion"
+              echo "    };"
+              echo "})"
+              echo "module.exports = {solidity: '${solc_version}'};"
+            } >> hardhat.config.js
+      - run:
+          name: Build and test the boilerplate project with local Hardhat
+          command: |
+            cd boilerplate/
+            npm run test
+
+  truffle-sample-project:
+    docker:
+      - image: circleci/node:12
+    steps:
+      - show_npm_version
+      - provision_and_package_solcjs
+      - provision_truffle_with_packaged_solcjs
+      - run:
+          name: Unbox MetaCoin
+          command: |
+            mkdir metacoin/
+            cd metacoin/
+            node ../truffle/node_modules/.bin/truffle unbox metacoin
+      - run:
+          name: Strip version pragmas
+          command: sed -i 's|pragma solidity [^;]\+;|pragma solidity *;|g' $(find metacoin/{contracts,test}/ -name "*.sol")
+      - run:
+          name: Build and test the sample project with local Truffle and its default solc
+          command: |
+            cd metacoin/
+            node ../truffle/node_modules/.bin/truffle test
+      - run:
+          name: Build and test the sample project with local Truffle and latest solc
+          command: |
+            cd metacoin/
+            # `truffle test` compiles the project but artifacts go into /tmp/
+            ! [[ -e build/ ]] || false
+            echo "module.exports['compilers'] = {solc: {version: '$(realpath ../solc-js/)'}}" > truffle-config.js
+            node ../truffle/node_modules/.bin/truffle test
+
+  solidity-solcjs-ext-test:
+    docker:
+      - image: circleci/node
+    steps:
+      - show_npm_version
+      - checkout:
+          path: solc-js/
+      - run: git clone --depth 1 "https://github.com/ethereum/solidity" solidity/
+      - run: cd solidity/ && curl "https://binaries.soliditylang.org/bin/soljson-nightly.js" --location --output soljson.js
+      - run: cd solidity/ && test/externalTests/solc-js/solc-js.sh "$(realpath soljson.js)" "$(scripts/get_version.sh)"
```

Comment:
I just realized that this will run with solc-js independently checked out from `master` by the `solc-js.sh` script. I need to adjust `solc-js.sh` in the main repo to accept an existing checkout of solc-js instead. For now I'm removing it from the PR and I'll resubmit it separately.